src/graphql/types/Advertisement/createdAt.ts

```diff
@@ -1,64 +1,78 @@
+import type { GraphQLContext } from "~/src/graphql/context";
 import { TalawaGraphQLError } from "~/src/utilities/TalawaGraphQLError";
 import envConfig from "~/src/utilities/graphqLimits";
-import { Advertisement } from "./Advertisement";
+import {
+	Advertisement,
+	type Advertisement as AdvertisementType,
+} from "./Advertisement";
 
-Advertisement.implement({
-	fields: (t) => ({
-		createdAt: t.field({
-			description: "Date time at the time the advertisement was created.",
-			complexity: envConfig.API_GRAPHQL_SCALAR_RESOLVER_FIELD_COST,
-			resolve: async (parent, _args, ctx) => {
-				if (!ctx.currentClient.isAuthenticated) {
-					throw new TalawaGraphQLError({
-						extensions: {
-							code: "unauthenticated",
-						},
-					});
-				}
-
-				const currentUserId = ctx.currentClient.user.id;
+export const createdAtResolver = async (
+	parent: AdvertisementType,
+	_args: Record<string, never>,
+	ctx: GraphQLContext,
+) => {
+	if (!ctx.currentClient.isAuthenticated) {
+		throw new TalawaGraphQLError({
+			extensions: {
+				code: "unauthenticated",
+			},
+		});
+	}
 
-				const currentUser = await ctx.drizzleClient.query.usersTable.findFirst({
-					columns: {
-						role: true,
-					},
-					with: {
-						organizationMembershipsWhereMember: {
-							columns: {
-								role: true,
-							},
-							where: (fields, operators) =>
-								operators.eq(fields.organizationId, parent.organizationId),
-						},
-					},
-					where: (fields, operators) => operators.eq(fields.id, currentUserId),
-				});
+	if (!ctx.currentClient.user?.id) {
+		throw new TalawaGraphQLError({
+			extensions: { code: "unauthenticated" },
+		});
+	}
+	const currentUserId = ctx.currentClient.user.id;
 
-				if (currentUser === undefined) {
-					throw new TalawaGraphQLError({
-						extensions: {
-							code: "unauthenticated",
-						},
-					});
-				}
+	const currentUser = await ctx.drizzleClient.query.usersTable.findFirst({
+		columns: {
+			role: true,
+		},
+		with: {
+			organizationMembershipsWhereMember: {
+				columns: {
+					role: true,
+				},
+				where: (fields, operators) =>
+					operators.eq(fields.organizationId, parent.organizationId),
+			},
+		},
+		where: (fields, operators) => operators.eq(fields.id, currentUserId),
+	});
 
-				const currentUserOrganizationMembership =
-					currentUser.organizationMembershipsWhereMember[0];
+	if (currentUser === undefined) {
+		throw new TalawaGraphQLError({
+			extensions: {
+				code: "unauthenticated",
+			},
+		});
+	}
 
-				if (
-					currentUser.role !== "administrator" &&
-					(currentUserOrganizationMembership === undefined ||
-						currentUserOrganizationMembership.role !== "administrator")
-				) {
-					throw new TalawaGraphQLError({
```

Comment:
Why was this logic changed when it worked before and was approved by Code Rabbit?